### PR TITLE
Update DisableIfDisplayName javadoc to match online documentation

### LINKED ISSUE #547: DisableIfDisplayName javadoc incorrect as of 1.5.0
As of 1.5.0, the online docs for `DisableIfDisplayName` have this [note](https://junit-pioneer.org/docs/disable-parameterized-tests/#disableifdisplayname):

> Since JUnit Pioneer 1.5.0, using both matches and contains in a single annotation is no longer permitted.

And it then goes on to provide additional background information.

But the javadoc in `DisableIfDisplayName` still says that both can be configured:

> If neither contains nor matches is configured, the extension will throw an exception. It is possible to configure both, in which case the test gets disabled if at least one substring was found or at least one regular expression matched.

I only found this when I updated a project to 1.5.0 and it had a test that was using _both_ `contains` and `matches` and the test failed with the message:

> @DisableIfDisplayName requires that either `contains` or `matches` is specified, but both are present.

### PR DESCRIPTION
**Pull request by sleberknight**
Change javadoc to state that either matches or contains must be
present, but not both.

Closes #547
PR #548 

---
**PR checklist**

The following checklist shall help the PR's author, the reviewers and maintainers to ensure the quality of this project.
It is based on our contributors guidelines, especially the ["writing code" section](https://github.com/junit-pioneer/junit-pioneer/blob/main/CONTRIBUTING.md#writing-code).
It shall help to check for completion of the listed points.
If a point does not apply to the given PR's changes, the corresponding entry can be simply marked as done. 

Documentation (general)
* [ ] There is documentation (Javadoc and site documentation; added or updated)
* [ ] There is implementation information to describe _why_ a non-obvious source code / solution got implemented
* [ ] Site documentation has its own `.adoc` file in the `docs` folder, e.g. `docs/report-entries.adoc`
* [ ] Only one sentence per line (especially in `.adoc` files)
* [ ] Javadoc uses formal style, while sites documentation may use informal style

Documentation (new extension)
* [ ] The `docs/docs-nav.yml` navigation has an entry for the new extension
* [ ] The `package-info.java` contains information about the new extension

Code
* [ ] Code adheres to code style, naming conventions etc.
* [ ] Successful tests cover all changes
* [ ] There are checks which validate correct / false usage / configuration of a functionality and there are tests to verify those checks
* [ ] Tests use [AssertJ](https://assertj.github.io/doc/) or our own [PioneerAssert](https://github.com/junit-pioneer/junit-pioneer/blob/main/CONTRIBUTING.md#assertions) (which are based on AssertJ)

Contributing
* [ ] A prepared commit message exists
* [ ] The list of contributions inside `README.md` mentions the new contribution (real name optional) 

---

I hereby agree to the terms of the [JUnit Pioneer Contributor License Agreement](https://github.com/junit-pioneer/junit-pioneer/blob/main/CONTRIBUTING.md#junit-pioneer-contributor-license-agreement).

### CODE REVIEW

**Comment by Michael1993** [comment]
Hi, thanks for submitting this (and for catching it in the first place)! We'd like to mention your contribution in the `README.md` if that's alright with you. You can add yourself in a commit or we (the maintainers) can do it for you before merging. Up to you, let us know.

Note, if you do add yourself, the names are in lexicographic order.

**Review by Michael1993 — APPROVED** [review]
Nice catch and nice fix. Thanks again!

**Comment by sleberknight** [comment]
Added contribution to README. Please feel free to change the wording if you want.

**Review by Bukama — APPROVED** [review]

**Pull request merged**
